rts/Lua/LuaVBOImpl.cpp

```diff
@@ -712,6 +722,14 @@ sol::as_table_t<std::vector<lua_Number>> LuaVBOImpl::Download(sol::optional<int>
 	return sol::as_table(dataVec);
 }
 
+void LuaVBOImpl::Clear()
+{
+	VBOExistenceCheck(vbo, __func__);
+
+	GLubyte val = 0;
+	glClearNamedBufferData(GetId(), GL_R8UI, GL_RED_INTEGER, GL_UNSIGNED_BYTE, &val);
```

Comment:
Please avoid any `gl*Named*` calls, usually only GL4.5/4.6 support them and we generally don't use them because of that. Use `glClearBufferData` instead.

Apart from that every new function such as `glClearBufferData` has to be added into `glstub.c` to support headless engine. See https://github.com/beyond-all-reason/spring/commit/304b4b25c8c39b1c38cb72e2decd769fcf6327b2 as example. Without such stubs the headless engine won't link.

Reply:
> Please avoid any `gl*Named*` calls, usually only GL4.5/4.6 support them and we generally don't use them because of that. Use `glClearBufferData` instead.
> 
> Apart from that every new function such as `glClearBufferData` has to be added into `glstub.c` to support headless engine. See [304b4b2](https://github.com/beyond-all-reason/spring/commit/304b4b25c8c39b1c38cb72e2decd769fcf6327b2) as example. Without such stubs the headless engine won't link.

Gotcha.
Shall add the stubs.
About old GL, I got an idea: since all Lua entries are pushed dynamically, if I wanted I could make a second version of the same function that could be wired if detected GL version has support for gl*Named* calls. That would not require much extra lines, because I would use a template, or just a condition check that is literally weightless.
Maybe it's worth the try?
This tiny function _won't give much benefit_, obviously, but we might be missing on much useful stuff **_later_** if we restrict ourselves and don't find ways to use the new stuff.

Reply:
I am not planning to write an unsafe code, but I am up to some experimentations in code design, if it allows to use new features.

Reply:
Some engine calls are done that way (supporting both old and new GL calls), but generally it's PITA to keep them in sync and usually a couple of extra calls to bind/unbind don't worth the effort as the bottlenecks are elsewhere. 

You're free to try, but I can't promise to merge that code, because readability and maintainability are important traits in our business.

Reply:
@lhog Done stubs and glClearBufferData fallback